src/solvers/mod.rs

```diff
@@ -234,3 +237,14 @@ pub trait ModelWithSOS1 {
         self
     }
 }
+
+/// A model that supports setting the MIP gap
+pub trait WithMipGap {
+    /// Get the relative MIP gap
+    fn mip_gap(&self) -> Option<f32>;
+
+    /// Set the relative MIP gap
+    fn with_mip_gap(self, mip_gap: f32) -> Result<Self, String>
```

Comment:
I think we should have a real error type here instead of a string. 

This can be a tiny enum with just two variants: Negative and Infinite. Then you should implement the standard error trait on this enum.

Reply:
Done (not pushed yet).  I included an Other(String) variant to handle any other error from the lp-solvers crate, which currently returns the error as a string (I could create an error type in that crate as well).

Reply:
Yes, I think an enum with Negative, Infinite, and Other is fine.